funnel/utils/markdown/base.py

```diff
@@ -0,0 +1,137 @@
+"""Base files for markdown parser."""
+
+from copy import deepcopy
+from typing import Any, List, Mapping, Optional, Type, Union, overload
+
+from markdown_it import MarkdownIt
+from markupsafe import Markup
+from mdit_py_plugins import anchors, footnote, tasklists
+from typing_extensions import Protocol
+
+from coaster.utils import make_name
+from coaster.utils.text import VALID_TAGS, normalize_spaces_multiline, sanitize_html
+
+MARKDOWN_HTML_TAGS = deepcopy(VALID_TAGS)
+
+
+# --- Typing Protocol for markdown-it-py plugins ---------------------------------------
+class MarkdownItPluginProtocol(Protocol):  # pylint: disable=too-few-public-methods
+    """Protocol for callable that gets a callback to initilize markdown-it-py plugin."""
+
+    def __call__(self, md: MarkdownIt, **options) -> None:
+        ...
+
+
+# --- Standard extensions --------------------------------------------------------------
+# FOR CUT 2
+# TODO: caret, tilde:
+#       ^^ins^^, ^sup^ dont work OOTB. ~~del~~ uses <s/>, not <del/>.
+#       Can port 1st 2 from markdown-it-[sup|ins] and implement del separately.
+#       Port from https://github.com/markdown-it/markdown-it-sup
+#       Port from https://github.com/markdown-it/markdown-it-ins
+# TODO: emoji, (mark => highlight, inlinehilite)
+#       Port from https://github.com/markdown-it/markdown-it-emoji
+#       Port from https://github.com/markdown-it/markdown-it-mark
+#       Evaluate:
+#       https://www.npmjs.com/search?q=highlight%20keywords%3Amarkdown-it-plugin
+
+
+default_markdown_extensions_html: Mapping[str, Type[MarkdownItPluginProtocol]] = {
```

Comment:
The `Type[]` wrapper should not be necessary. You can use `Dict[str, MarkdownItPluginProtocol]` as this is an explicit dict and not an abstract Mapping.